main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/domainr/whois"
+	"github.com/jessevdk/go-flags"
+)
+
+// Options with all cli commands and flags
+type Options struct {
+	Path     string `long:"config" short:"c" env:"WOD_CONFIG" description:"Path to config.json" default:"./config.json"`
+	Dest     string `long:"destination" short:"d" env:"WOD_DESTINATION" description:"Path to destination directory" default:"./web"`
+	Template string `long:"template" short:"t" env:"WOD_TEMPLATE_PATH" description:"Path to html template" default:"./template.html"`
+}
+
+// Domain struct
+type Domain struct {
+	Name        string `json:"name"`
+	PricePerDay int    `json:"price_per_day"`
+	Rate        int    `json:"rate"`
+}
+
+// Config struct
+type Config struct {
+	Domains     []Domain `json:"domains"`
+	PricePerDay int      `json:"price_per_day"`
+	Rate        int      `json:"rate"`
+}
+
+// TemplateData struct
+type TemplateData struct {
+	Domain          string
+	Price           int
+	RegistratedDate string
+}
+
+func main() {
```

Comment:
Unit tests are missing, they are very useful and it's uncommon to have Go code without such tests. Tests are specifications on how your code works and allow a developer to introduce changes to the code with a decent understanding of what is changed in program behavior.

Also as soon as you start writing tests you would understand why current amount of code in main function is too big and which parts should be moved out of it.